<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>


  <arg name="robotID" default="1"/>
  <arg name="NUM_ROBOTS" default="2" />
  <arg name="NEIGHBOR_TOPIC" default="/throttledPose"/>
  <arg name="usingSimulation" default="0" />

  <group ns="machine_$(arg robotID)">
  <node name="nmpc_planner_$(arg robotID)" pkg="nmpc_planner" type="mpc_DQMPC" args="$(arg robotID) $(arg NUM_ROBOTS)" output="screen">
    <param name="uavPoseTopic" value="/machine_" type="string" />
    <param name="uavPoseTopicSuffix" value="/pose" type="string" />
    <param name="uavNeighborTopicSuffix" value="$(arg NEIGHBOR_TOPIC)" type="string" />

    <param name="uavPoseTrajTopic" value="/machine_" type="string" />
    <param name="uavPoseTrajTopicSuffix" value="/mpc_calculated/pose_traj" type="string" />
    <param name="uavOffsetTopic" value="/machine_$(arg robotID)/target_tracker/offset" type="string" />

    <param name="outputPoseTopic" value="/machine_$(arg robotID)/command" type="string" />

    <param name="objectGTTopic" value="/machine_$(arg robotID)/target_tracker/pose" type="string" />
    <param name="objectGTVelTopic" value="/machine_$(arg robotID)/target_tracker/twist" type="string" />
    <param name="objectEstimatedTopic" value="/machine_$(arg robotID)/self_target_tracker/pose" type="string" />
    <param name="objectEstimatedVelTopic" value="/machine_$(arg robotID)/self_target_tracker/twist" type="string" />

    <param name="obstacleTopicBase" value="NewObstaclePoints_$(arg robotID)" type="string" />
    <param name="NMPC_timesteps" value="3" type="int" />

    <param name="useGTforTarget" value="true" type="bool" />
    <param name="deltaT" value="0.1" type="double" />

    <param name="usingSimulation" value="$(arg usingSimulation)" type="bool" />

    <param name="INTERNAL_SUB_STEP" value="0.1" type="double" />

    <param name="targetRepulsionGain" value="50" type="double" />
    <param name="tFormationRepulGain" value="35.0" type="double" />
    <param name="tGoalAttractionGain" value="7.0" type="double" />


    <param name="neighborDistThreshold" value="1.0" type="double" />
    <param name="maxOffsetUncertaintyRadius" value="3.0" type="double"/>
    <param name="distanceThresholdToTarget" value="8.0" type="double" />
    <param name="copterDesiredHeightinNED" value="-8.0" type="double" />
    <param name="POINT_OBSTACLES" value="true" type="bool" />
    <param name="usingSimulation" value="$(arg usingSimulation)" type="bool" />


    <param name="activeTrackingWeight" value="100.0" type="double" />
    <param name="energyWeight" value="10" type="double" />

    <param name="copterPositionLimitX" value="20.0" type="double" />
    <param name="copterPositionLimitY" value="20.0" type="double" />
    <param name="copterPositionLowerLimitZ" value="-20.0" type="double" />
    <param name="copterPositionUpperLimitZ" value="-1.0" type="double" />

    <param name="copterVelocityLimitHorizontal" value="5" type="double" />
    <param name="copterVelocityLimitVertical" value="0.5" type="double" />

    <param name="copterAccelarationLimitHorizontal" value="5.0" type="double" />
    <param name="copterAccelarationLimitVertical" value="11.0" type="double" />

    <param name="useZeroAsFixedTarget" value="false" type="bool" />

    <rosparam command="load" file="$(find nmpc_planner)/cfg/obstacles_simulation.yaml" />

   </node>
   </group>

</launch>
